### ISSUE TITLE
V2 mentors feedback for existing solutions

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/discussions#189]
With nextercism, part of its appeal is the dedicated mentors team that will be reviewing submissions. My question is simple: will they also review any existing submissions? I'm asking this because there have been tracks where I received hardly any feedback, and that sort of defeats the point of Exercism :)

**Comment by NobbZ** [comment]
But reviewing all the old stuff will take ages… In the first run all the old things were imported as unmentored, and mentors got a 4 year backlog… You can't work through that… Also you do not even know if those students are still interested in exercism or feedback…

**Comment by ErikSchierboom** [comment]
@NobbZ That's true, but you do see my point, do you? Perhaps have some option for people to re-submit solutions? That way you only have to review the solutions for those people that really care about it.

I'm not saying we should do this, but I am saying it is worth discussing.

**Comment by NobbZ** [comment]
Yeah, and to be homest, I thought about how to do it best already…

Perhaps consider every old solution as “approved by age” and “published”, until the “owner” of the submissions logs in the first time. Give him a modal if he is interested in his old stuff getting back into the review process, or if he want to keep them published or if he wants to unpublish them…

Then of course do as the “owner” requests.

**Comment by ErikSchierboom** [comment]
That sounds like a reasonable plan. @kytrinyx What do you think?

**Comment by Insti** [comment]
Meh. Only nextercism review new iterations, if you want an old one looked at re-submit it.
make a minor change if needed to dodge the duplicate detector.

**Comment by NobbZ** [comment]
Nah… Thats the worst… If you do it like this, we need basically not ETL at all… Then we could just mark every single exercise as solved or not, dropping all those space consuming code from the past…

**Comment by Insti** [comment]
What is ETL?

Am I confusing the issue here?

Previous solutions should still be assessable, just not added to the mentor queue.

**Comment by NobbZ** [comment]
ETL is **E**xtract-**T**ransform-**L**oad, basically extracting data from an old database, transforming it that the new database can deal with it and loading it into the new database.

In your scenaria a student that really needs the help but submits in the very last seconds of V1, does not get any mentoring, nor is he guided to get it, while another person who does not need help, but submits 10 second late, will be blocked for some time until he gets approved.

Therefore I think we need an easy way for students to opt-in for getting a mentor easily without going through the hazzle of resubmitting stuff.

**Comment by Insti** [comment]
Ok thanks.
I'm not against your proposal. :heart:
But I think that changing the mentor queue to include solutions submitted in the last n days and ignoring anything older would be sufficient and easier to implement than an interface that requires user interaction.

**Comment by kytrinyx** [comment]
The ETL is going to auto-approve and publish (to the Exercism community's pool of solutions to browse, not to people's profile, that's an explicit choice) anything older than 2 weeks, but we're going to give people the option to "un-publish" exercises.

My goal is to make a way for people to specifically request mentorship for old exercises—but only a handful at a time on any given track, so as to not overload the mentors.

We've got on our list to discuss this in detail when @iHiD gets back from Barcelona.

**Comment by tleen** [comment]
This may not be directly related, but it may be useful to have some sort of tool that keeps an eye on the mentor queue depth so we can get a heads up if it is getting really long for specific tracks. That may be a good indicator that it is not being mentored or the mentor(s) are AFK for an extended period of time.

**Comment by kytrinyx** [comment]
>  it may be useful to have some sort of tool that keeps an eye on the mentor queue depth

Yes, I think that would be wise. We've got a lot of work to do around mentorship to make sure that each track has the support it needs.

I also have to kick off a discussion of what the right thing to do is if we don't have enough mentors to properly support a track. My head is too full to think about it right now, but it's on my list of critical things to figure out before we can launch.

**Comment by tleen** [comment]
> I also have to kick off a discussion of what the right thing to do is if we don't have enough mentors to properly support a track

I'd be happy to corral this discussion for you if it would be helpful.

**Comment by kytrinyx** [comment]
@tleen you would be my hero. I'm going to email you a brain dump if that's OK, and then if you would figure out what is useful to post from that this would be so very helpful.
